apps/backend/src/mcp/tools/device_analysis.py

```
@@ -0,0 +1,491 @@
+                results["docker_info"]["docker_info"] = docker_info
+                
+                # Detect Docker Compose projects and common paths
+                compose_result = await execute_ssh_command_simple(
+                    device,
+                    "find /home /opt /srv -name 'docker-compose.yml' -o -name 'docker-compose.yaml' 2>/dev/null | head -10",
+                    timeout=20
+                )
+                
+                if compose_result.return_code == 0:
+                    compose_files = [f.strip() for f in compose_result.stdout.strip().split('\n') if f.strip()]
+                    compose_base_paths = list(set(['/'.join(f.split('/')[:-1]) for f in compose_files]))
```

Ruff (0.12.2)

Unnecessary list comprehension (rewrite as a set comprehension)

Rewrite as a set comprehension

(C403)